packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-oidc.tsx

```diff
@@ -0,0 +1,167 @@
+  Description,
+  FormFieldContainer,
+  Label,
+  TextInput,
+} from '@mongodb-js/compass-components';
+import type ConnectionStringUrl from 'mongodb-connection-string-url';
+import type { ConnectionOptions } from 'mongodb-data-service';
+import type { AuthFlowType } from '@mongodb-js/oidc-plugin';
+
+import type { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import type { ConnectionFormError } from '../../../utils/validation';
+import { errorMessageByFieldName } from '../../../utils/validation';
+import {
+  getConnectionStringUsername,
+  parseAuthMechanismProperties,
+} from '../../../utils/connection-string-helpers';
+
+type OIDCOptions = NonNullable<ConnectionOptions['oidc']>;
+
+function AuthenticationOIDC({
+  connectionStringUrl,
+  updateConnectionFormField,
+  errors,
+  connectionOptions,
+}: {
+  connectionStringUrl: ConnectionStringUrl;
+  errors: ConnectionFormError[];
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionOptions: ConnectionOptions;
+}): React.ReactElement {
+  const username = getConnectionStringUsername(connectionStringUrl);
+  const usernameError = errorMessageByFieldName(errors, 'username');
+
+  const handleFieldChanged = useCallback(
+    (key: keyof OIDCOptions, value?: OIDCOptions[keyof OIDCOptions]) => {
+      return updateConnectionFormField({
+        type: 'update-oidc-param',
+        key: key,
+        value,
+      });
+    },
+    [updateConnectionFormField]
+  );
+
+  const authMechanismProperties =
+    parseAuthMechanismProperties(connectionStringUrl);
+  const allowedHosts = authMechanismProperties.get('ALLOWED_HOSTS');
+  const hasEnabledUntrustedEndpoints = allowedHosts === '*';
+
+  const hasEnabledDeviceAuthFlow = !!(
+    connectionOptions.oidc?.allowedFlows as AuthFlowType[]
+  )?.includes?.('device-auth');
+
+  return (
+    <>
+      <FormFieldContainer>
+        <TextInput
+          data-testid="connection-oidc-username-input"
+          onChange={({
+            target: { value },
+          }: React.ChangeEvent<HTMLInputElement>) => {
+            updateConnectionFormField({
+              type: 'update-username',
+              username: value,
+            });
+          }}
+          label="Principal"
+          optional
+          value={username || ''}
+          errorMessage={usernameError}
+          state={usernameError ? 'error' : undefined}
+        />
+      </FormFieldContainer>
+      <FormFieldContainer>
+        <Accordion text="OIDC Options">
+          <FormFieldContainer>
+            <TextInput
+              data-testid="connection-oidc-auth-code-flow-redirect-uri-input"
+              onChange={({
+                target: { value },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                handleFieldChanged('redirectURI', value);
+              }}
+              optional
+              label="Auth Code Flow Redirect URI"
+              value={connectionOptions?.oidc?.redirectURI || ''}
+            />
```

Comment:
Do we want to add a short description here similar to the one below? E.g. “This value needs to match the configuration of the Identity Provider used by the server.”